.github/workflows/tests-e2e-upgrade-restart.yaml

```diff
@@ -0,0 +1,816 @@
+          fi
+
+          cilium status --wait
+          kubectl get pods --all-namespaces -o wide
+          kubectl -n kube-system exec daemonset/cilium -- cilium status
+
+          mkdir -p cilium-junits
+
+      - name: Install node local DNS
+        if: ${{ matrix.node-local-dns == 'true' }}
+        shell: bash
+        run: |
+          kubedns=$(kubectl get svc kube-dns -n kube-system -o jsonpath={.spec.clusterIP}) && sed -i "s/__PILLAR__DNS__SERVER__/$kubedns/g;" untrusted/cilium-newest/examples/kubernetes-local-redirect/node-local-dns.yaml
+          sed -i "s/__PILLAR__UPSTREAM__SERVERS__/1.1.1.1/g;" untrusted/cilium-newest/examples/kubernetes-local-redirect/node-local-dns.yaml
+          kubectl apply -k untrusted/cilium-newest/examples/kubernetes-local-redirect
+          kubectl rollout status -n kube-system ds/node-local-dns
+
+      - name: Start conn-disrupt-test
+        shell: bash
+        run: |
+          # Create pods which establish long lived connections. It will be used by
+          # subsequent connectivity tests with --include-conn-disrupt-test to catch any
+          # interruption in such flows.
+          cilium connectivity test --include-conn-disrupt-test --conn-disrupt-test-setup \
+            --conn-disrupt-test-restarts-path "./cilium-conn-disrupt-restarts" \
+            --conn-disrupt-dispatch-interval 0ms
+
+      - name: Upgrade Cilium
+        if: ${{ matrix.skip-upgrade != 'true' }}
+        shell: bash
+        run: |
+          cilium upgrade \
+            ${{ steps.cilium-newest-config.outputs.config }} \
+            ${{ steps.kvstore.outputs.config }}
+
+          cilium status --wait --wait-duration=10m
+          kubectl get pods --all-namespaces -o wide
+          kubectl -n kube-system exec daemonset/cilium -- cilium status
+
+      - name: Setup Cilium CLI flags
+        id: cli-flags
+        run: |
+          EXTRA=()
+          if [ "${{ matrix.secondary-network }}" = "true" ]; then
+            EXTRA+=("\"--secondary-network-iface=eth1\"")
+          fi
+
+          if [ "${{ matrix.encryption-strict-mode }}" = "true" ]; then
+            # "Test Cilium after upgrade" ran strict-mode-encryption test which caused temporary drops.
+            EXTRA+=("\"--expected-drop-reasons=+Traffic is unencrypted\"")
+          fi
+
+          echo flags="--include-unsafe-tests \
+            --collect-sysdump-on-failure \
+            --flush-ct \
+            --sysdump-hubble-flows-count=1000000 \
+            --sysdump-hubble-flows-timeout=5m \
+            --sysdump-output-filename \"cilium-sysdump-${{ matrix.name }}-<ts>\" \
+            --junit-file \"cilium-junits/${{ env.job_name }} (${{ matrix.name }}).xml\" \
+            --junit-property github_job_step=\"Run tests upgrade 2 (${{ matrix.name }})\" \
+            ${EXTRA[@]}" >> $GITHUB_OUTPUT
+
+      - name: Restart kind nodes
+        uses: ./.github/actions/lvh-kind-restart
```

Comment:
Will this directly invoke the copy of the workflow in the branch, and hence also directly invoke the shell script from the branch? The reason I ask is that per the comment on lines 552-553, privileged workflows should not execute arbitrary shell scripts from the submitted branch. I think that [this blog](https://karimrahal.com/2023/01/05/github-actions-leaking-secrets/) may give a sense of the sorts of threats we're trying to protect against.

/cc @cilium/security

Reply:
> Will this directly invoke the copy of the workflow in the branch

This seems like a solid probably. I could go ahead and move the reboot logic into the workflow file itself and out of `contrib/scripts/kind-restart.sh` which should avoid the security concerns.